File: mobile/AAiT-mobile-group-4/e_commerce_app_clean/lib/features/authentication/domain/entities/sign_up.dart
import 'package:equatable/equatable.dart';

class SignUpEntity extends Equatable{
  final String email;
  final String password;
  final String username;

  const SignUpEntity({
    required this.email,
    required this.password,
    required this.username,
  });
  @override
  List<Object?> get props =>[
    email,
    password,
    username,
  ] ;
}

File: backend/AASTU-backend-group-4/domain/signup.go
package domain

type SignupRequest struct {
	Firstname string `json:"firstname" bson:"firstname" binding:"required"`
	Lastname  string `json:"lastname" bson:"lastname" binding:"required"`
	Username  string `json:"username" bson:"username" binding:"required"`
	Password  string `json:"password" bson:"password" binding:"required"`
	Email     string `json:"email" bson:"email" binding:"required"`
}

type SignupResponse struct {
	AccessToken string `json:"accessToken"`
}
